lnwallet/btcwallet/btcwallet.go

```diff
@@ -1202,7 +1202,8 @@ func mapRpcclientError(err error) error {
 	switch {
 	// If the wallet reports a double spend, convert it to our internal
 	// ErrDoubleSpend and return.
-	case errors.Is(err, rpcclient.ErrMempoolConflict),
```

Comment:
Why is `ErrMempoolConflict` handling removed?

Reply:
I think we only get this error when,
1) try to fee bump a tx when it doesn't signal RBF
2) try to spend the same output in a different tx

Either way it shouldn't ever happen in ln. However I decided to not change it now to focus on fixing the error matching issue.